src/analyzer.ts

```diff
@@ -52,8 +52,8 @@ export interface Options {
 
 export class NoKnownParserError extends Error {};
 
-export interface ScannerTable extends Map<string, Scanner<any, any, any>[]> {}
-export interface LazyEdgeMap extends Map<string, string[]> {}
+export class ScannerTable extends Map<string, Scanner<any, any, any>[]> {}
+export class LazyEdgeMap extends Map<string, string[]> {}
```

Comment:
I think what you probably want here is:

```typescript
export type ScannerTable = Map<string, Scanner<any, any, any>[]>;
export type LazyEdgeMap = Map<string, string[]> {};
```

If they're classes then they'll exist at runtime and you'll be able to `new` them and such, which I don't think we want here.